.github/workflows/builder_docker-based_slsa3.yml

```
@@ -143,90 +153,118 @@ jobs:
     # TODO: would it be convenient to output the builderDependency?
     # TODO: this is a no-op right now. Replace with final builder.
     outputs:
-      builder-binary-sha256: ${{ steps.generate-builder.outputs.sha256 }}
-    needs: detect-env
+      builder-binary-sha256: ${{ steps.generate.outputs.sha256 }}
+    needs: [detect-env, rng]
     runs-on: ubuntu-latest
     steps:
-      - name: Generate the builder binary
-        id: generate-builder
-        run: |
-          echo "fetches the builder binary and uploads it"
-          echo "sha256=5b36fa157544ece32dba62c14480d85322b86918032ca5f40b87f5d1dc386d39" >> $GITHUB_OUTPUT
+      - name: Generate builder binary
+        id: generate
+        uses: slsa-framework/slsa-github-generator/.github/actions/generate-builder@main
+        with:
+          repository: "${{ needs.detect-env.outputs.repository }}"
+          ref: "${{ needs.detect-env.outputs.ref }}"
+          go-version: 1.19
+          # Note: This must be the non-randomized binary name, so that it can be downloaded from the release assets.
+          binary: "${{ env.BUILDER_BINARY }}"
+          compile-builder: "${{ inputs.compile-builder }}"
+          directory: "${{ env.BUILDER_DIR }}"
+          allow-private-repository: ${{ inputs.private-repository }}
+
+      - name: Upload builder
+        uses: actions/upload-artifact@83fd05a356d7e2593de66fc9913b3002723633cb # tag=v3.1.1
```

Renovate doesn't seem to use `tag=` anymore.
```suggestion
        uses: actions/upload-artifact@83fd05a356d7e2593de66fc9913b3002723633cb # v3.1.1
```